Automattic/pocket-casts-android
# Widget - Support dynamic colors

## Description

This adds support for dynamic colors for the player widget. 

## Testing Instructions
1. Install the app 
2. Long press the app icon 
3. Tap Widgets
4. Drag the widget to the home screen
5. ✅ Notice that the colors of the button matches those of the wallpaper
6. Change the wallpaper
7. ✅ Notice that the colors of the widget are updated

** Test on Android 11, Android 12, and Android 13 devices (light + dark theme)
** Dynamic colors are not supported below Android 12 so original colors are retained on versions less than Android 12.

## Screenshots or Screencast 

https://github.com/Automattic/pocket-casts-android/assets/1405144/6c59d311-3f44-45f1-90e6-a3395e08d652


Android 9 (API 28) | Android 12 | Android 14 (light theme)
---|---|---
<img width=200 src="https://github.com/Automattic/pocket-casts-android/assets/1405144/ad6dbe91-5a0f-4920-9df4-91d0b3a92a96"/> | <img width=200 src="https://github.com/Automattic/pocket-casts-android/assets/1405144/16fd5052-f363-4f32-94d6-5c1832de3c5c"/> | <img width=200 src="https://github.com/Automattic/pocket-casts-android/assets/1405144/f05081a3-3f5a-417f-aceb-0f8a0b8a4cdf"/>


## Checklist
 
#### I have tested any UI changes...
